src/snovault/schema_utils.py

```diff
@@ -281,6 +282,19 @@ def notSubmittable(validator, linkTo, instance, schema):
     yield ValidationError('submission disallowed')
 
 
+# modifying https://github.com/lrowe/jsonschema_serialize_fork/blob/master/jsonschema_serialize_fork/_validators.py#L143
+def pattern(validator, patrn, instance, schema):
+    if isinstance(patrn, list):
+        patrn = ''.join(patrn)
+
+    if (
+        validator.is_type(instance, "string") and
+        not re.search(patrn, instance)
+    ):
+        yield ValidationError("%r does not match %r" % (instance, patrn))
```

Comment:
Should definitely add unit/integration test for this.

Probably easiest thing to do would be to modify or add your own fake testing schema here: 
https://github.com/Lattice-Data/snovault/blob/d6bec11aa60244172d52512d918fc65663071073/src/snovault/tests/testing_views.py#L277

And then try to post to the endpoint you made with different objects that either pass or fail pattern validation, like here but without the fixture: https://github.com/Lattice-Data/snovault/blob/d6bec11aa60244172d52512d918fc65663071073/src/snovault/tests/test_post_put_patch.py#L78

Or follow this approach and call validate explicitly: https://github.com/Lattice-Data/snovault/blob/d6bec11aa60244172d52512d918fc65663071073/src/snovault/tests/test_schema_utils.py#L17

Can see more examples of testing schema validation on real encoded file objects: https://github.com/ENCODE-DCC/encoded/blob/39478a6bff4dbc1c65c1fd889ffaa0ff0d74d557/src/encoded/tests/test_schema_file.py#L16